SignalExtraction/fitWm.C

```diff
@@ -744,167 +746,196 @@ void fitWm(const TString  outputDir,   // output directory
+              if(fabs(genVy)<0.5)
+                recoilCorr05->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              else if (fabs(genVy)>=0.5 && fabs(genVy)<1.0)
+                recoilCorr051->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              else
+                recoilCorr1->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0); 
+              // Compute the corrected MET value
+              TVector2 vMetCorr((corrMet)*cos(corrMetPhi),(corrMet)*sin(corrMetPhi));
+              Double_t corrMetWithLepton = (vMetCorr + vLepRaw - vLepCorr).Mod();
+              if(typev[ifile]==eWmunu)
+              {
+                hWmunuMetp->Fill(corrMetWithLepton,weight); corrMet=met, corrMetPhi=metPhi;
+              }
+              else
+              {
+                hEWKMetp->Fill(corrMetWithLepton,weight); corrMet=met, corrMetPhi=metPhi; 
+              }
+              //recoilCorrUp->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+              hWmunuMetp_PileupUp->Fill(corrMetWithLepton,weightUp); corrMet=met, corrMetPhi=metPhi;
+              //recoilCorrDown->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+              hWmunuMetp_PileupDown->Fill(corrMetWithLepton,weightDown);
               corrMet=met, corrMetPhi=metPhi;
             } else {
-              pU1 = 0; pU2 = 0; 
-              hWmunuMetm_RecoilDown->Fill(corrMet,weight);
+              //recoilCorrm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+              if(fabs(genVy)<0.5)
+                recoilCorrm05->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              else if (fabs(genVy)>=0.5 && fabs(genVy)<1.0)
+                recoilCorrm051->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0);
+              else
+                recoilCorrm1->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lep->Pt(),lep->Phi(),pU1,pU2,0); 
+              //recoilCorrm->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,genLepPt,genLepPhi,pU1,pU2,0);
+              // Compute the corrected MET value
+              TVector2 vMetCorr((corrMet)*cos(corrMetPhi),(corrMet)*sin(corrMetPhi));
+              Double_t corrMetWithLepton = (vMetCorr + vLepRaw - vLepCorr).Mod();
+              if(typev[ifile]==eWmunu)
+              {
+                hWmunuMetm->Fill(corrMetWithLepton,weight); corrMet=met, corrMetPhi=metPhi;
+              }
+              else
+              {
+                hEWKMetm->Fill(corrMetWithLepton,weight); corrMet=met, corrMetPhi=metPhi; 
+              }
+              //recoilCorrmUp->CorrectInvCdf(corrMet,corrMetPhi,genVPt,genVPhi,lepPt,lep->Phi(),pU1,pU2,0);
+              hWmunuMetm_PileupUp->Fill(corrMetWithLepton,weightUp);corrMet=met, corrMetPhi=metPhi;
```

Comment:
This histogram need to be populated only then the if(typev[ifile]==eWmunu) otherwise add the hEWKMetm_PileupUp same apply to the hEWKMetm_PileupDown